clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -564,6 +558,12 @@ void CodeGenFunction::FinishFunction(SourceLocation EndLoc) {
       ReturnValue = Address::invalid();
     }
   }
+
+  if (CGM.shouldEmitConvergenceTokens()) {
+    ConvergenceTokenStack.pop_back();
+    assert(ConvergenceTokenStack.empty() &&
+           "mismatched push/pop in convergence stack!");
+  }
```

Comment:
This does not seem to be related to the `hlsl_private` address space change and should be done in a separate PR (including the `GlobalConstructors.hlsl` test update).